config.c

```diff
@@ -105,25 +105,15 @@ janus_config *janus_config_parse(const char *config_file) {
 		/* Strip comments */
 		char *line = line_buffer, *sc = line, *c = NULL;
 		while((c = strchr(sc, ';')) != NULL) {
-			if(c == line) {
-				/* Comment starts here */
-				*c = '\0';
-				break;
-			}
-			c--;
-			if(*c != '\\') {
+			if(c == line || *(c-1) != '\\') {
 				/* Comment starts here */
 				*c = '\0';
```

Comment:
the real bug is that this would do `*c = '\0';` just after `c--;` so it would get the character before the semicolon